### PR TITLE
Introduce UManaged to model a span being a resource in Telemetry module

### PR DESCRIPTION
**Pull request by arjun-1**
As suggested here https://github.com/zio/zio-telemetry/issues/130, indeed Spans can be seen as resources and in that case they can and should be modeled using a `Managed`

### CODE REVIEW

File: opentelemetry/src/main/scala/zio/telemetry/opentelemetry/Tracing.scala [hunk 1]
```diff
@@ -17,8 +18,9 @@ object Tracing {
   trait Service {
     private[opentelemetry] def currentNanos: UIO[Long]
     private[opentelemetry] val currentSpan: FiberRef[Span]
-    private[opentelemetry] def createRoot(spanName: String, spanKind: Span.Kind): UIO[Span]
-    private[opentelemetry] def createChildOf(parent: Span, spanName: String, spanKind: Span.Kind): UIO[Span]
+    private[opentelemetry] def createRoot(spanName: String, spanKind: Span.Kind): UManaged[Span]
+    private[opentelemetry] def createChildOf(parent: Span, spanName: String, spanKind: Span.Kind): UManaged[Span]
+    private[opentelemetry] def end(span: Span): UIO[Unit]
```
Comment (runtologist) [inline opentelemetry/src/main/scala/zio/telemetry/opentelemetry/Tracing.scala:23]: If Spans are managed, why do we need to expose `end`?

File: opentelemetry/src/main/scala/zio/telemetry/opentelemetry/Tracing.scala [hunk 2]
```diff
@@ -27,29 +29,26 @@ object Tracing {
...
   }
 
   /**
-   * Sets the `currentSpan` to `span` only while `effect` runs.
-   * Then ends `span` according to the result of `effect`.
+   * Sets the `currentSpan` to `span` only while `effect` runs,
+   * and error status of `span` according to any potential failure of effect.
    */
-  private def finalizeSpanUsingEffect[R, E, A](
+  private def setSpanError[R, E, A](
```
Comment (runtologist) [inline opentelemetry/src/main/scala/zio/telemetry/opentelemetry/Tracing.scala:51]: Nit: As this does not only set an error status, I liked the previous name better.

File: opentelemetry/src/main/scala/zio/telemetry/opentelemetry/Tracing.scala (hunk 1 above)
Comment (arjun-1) [reply opentelemetry/src/main/scala/zio/telemetry/opentelemetry/Tracing.scala]: Good point! We don't need to indeed

Fixed with https://github.com/zio/zio-telemetry/pull/136/commits/5189adc8ba1a577ab76892b81bfa834ce9c70551

File: opentelemetry/src/main/scala/zio/telemetry/opentelemetry/Tracing.scala (hunk 2 above)
Comment (arjun-1) [reply opentelemetry/src/main/scala/zio/telemetry/opentelemetry/Tracing.scala]: No problem :). Fixed with https://github.com/zio/zio-telemetry/pull/136/commits/1d528cedd20b3bd7fc688cc7d7d58ec429e7bdaf

**Pull request merged**